wrappers/src/app_cs.cpp

```diff
@@ -87,7 +87,7 @@ namespace realm {
             void* managed_http_client;
         };
 
-        class SyncLogger : public util::RootLogger {
+        class SyncLogger : public util::Logger {
```

Comment:
Do we need to change the logger factory as well?

Reply:
I am not completely sure if we need to do any changes to our logger factory or not given what has been done in https://github.com/realm/realm-core/pull/5980. There is a comment a little bit up in which I invoked Yavor for suggestions 😁

Reply:
Looking at the c api changes, it seems to be a matter of replacing `make_unique` with `make_shared`: https://github.com/realm/realm-core/pull/5980/files#diff-317d315bdd0e8e2bc5b3bc3a914dcaea2ea477eb8f96b05a9a8b7dfa2ee6e980R61-R62

Reply:
I've pushed a commit for this, hoping it makes sense